airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_options/request_options_provider.py

```diff
@@ -20,7 +20,7 @@ class RequestOptionsProvider(ABC):
     """
 
     @abstractmethod
-    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+    def get_request_params(self, **kwargs) -> MutableMapping[str, Any]:
```

Comment:
Due to the collision between the `request_params, `request_headers` etc. dataclass fields, and the existing method names, this requires that we rename the underlying methods. The alternative is changing the language grammar, but that has bigger UX implications compared to just changing the interface under the hood.